### ISSUE TITLE
Wrong foreign key specification generated in MySQL, when foreign key shares name with unique key in different table

### ISSUE THREAD

**Issue by tobys-at-canva** [jOOQ/jOOQ#8270]
### Expected behavior and actual behavior:

Note: identical issue to https://github.com/jOOQ/jOOQ/issues/3520 but for MySQL (that appears to have been only fixed for Postgres).

This schema:

```
CREATE TABLE t_986_1 {
  ref INT,
  label CHAR(3),
  CONSTRAINT label_ref UNIQUE (label, ref),
  CONSTRAINT fk_986 FOREIGN KEY (ref) REFERENCES ref (_id)
}

CREATE TABLE t_986_2 (
  ref INT,
  CONSTRAINT fk_986 FOREIGN KEY (ref) REFERENCES ref (_id)
)
```

**Expected behaviour:**
- Produces foreign key column info with one field per FK.
- Order of fields passed to `Internal.createForeignKey(...)` in generated code is deterministic.

**Actual behaviour:**
- produces duplicate foreign key column information in the generated Keys.java class.
- Order of fields passed to `Internal.createForeignKey(...)` is non-deterministic.

### Steps to reproduce the problem:

- Create simple schema as outlined above.
- Generate code. 
- Optional: generate code multiple times to demonstrate non-determinism - but that seems secondary.

Sorry, I can't create a proper MCVE right now.

### Versions:

- jOOQ: 3.11.1
- Java: openjdk version "10.0.2" 2018-07-17
- Database (include vendor): MySQL
- OS: Ubuntu
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Thank you very much for your report. I'm not sure how to reproduce this case.

1. MySQL doesn't allow two constraints to have the same name. So, your second table definition will not pass. There should be an error: "Error Code: 1826. Duplicate foreign key constraint name 'fk_986'"
2. Would you mind posting the DDL of the `ref` table as well?

Thanks

**Comment by tobys-at-canva** [comment]
I was able to create an MCVE https://github.com/tobys-at-canva/jOOQ-mcve

There were a few issues in my example above. 

MySQL:

```
CREATE TABLE mcve.test (
  id    INT NOT NULL AUTO_INCREMENT PRIMARY KEY,
  value INT
);

CREATE TABLE mcve.t_1 (
  ref INT,
  value INT,
  CONSTRAINT idx_value UNIQUE (value, ref),
  CONSTRAINT idx_ref FOREIGN KEY (ref) REFERENCES test (id)
);

CREATE TABLE mcve.t_2 (
  value INT,
  CONSTRAINT idx_value FOREIGN KEY (value) REFERENCES test (id)
);
```

Produced these in `Keys.java`

```
    private static class UniqueKeys0 {
        public static final UniqueKey<FlywaySchemaHistoryRecord> KEY_FLYWAY_SCHEMA_HISTORY_PRIMARY = Internal.createUniqueKey(FlywaySchemaHistory.FLYWAY_SCHEMA_HISTORY, "KEY_flyway_schema_history_PRIMARY", FlywaySchemaHistory.FLYWAY_SCHEMA_HISTORY.INSTALLED_RANK);
        public static final UniqueKey<TestRecord> KEY_TEST_PRIMARY = Internal.createUniqueKey(Test.TEST, "KEY_test_PRIMARY", Test.TEST.ID);
        public static final UniqueKey<T_1Record> KEY_T_1_IDX_VALUE = Internal.createUniqueKey(T_1.T_1, "KEY_t_1_idx_value", T_1.T_1.VALUE, T_1.T_1.REF);
    }

    private static class ForeignKeys0 {
        public static final ForeignKey<T_1Record, TestRecord> IDX_REF = Internal.createForeignKey(org.jooq.mcve.Keys.KEY_TEST_PRIMARY, T_1.T_1, "idx_ref", T_1.T_1.REF);
        public static final ForeignKey<T_2Record, TestRecord> IDX_VALUE = Internal.createForeignKey(org.jooq.mcve.Keys.KEY_TEST_PRIMARY, T_2.T_2, "idx_value", T_2.T_2.VALUE, T_2.T_2.VALUE);
    }
```

**Comment by tobys-at-canva** [comment]
Note: I wasn't able to demonstrate the non-determinism we saw just yet but this seems like a good start.

**Comment by lukaseder** [comment]
Thank you very much. I'll try to reproduce this ASAP

**Comment by lukaseder** [comment]
Thanks for your patience. I can reproduce this `IDX_VALUE` references `T_2.VALUE` twice. Will investigate

**Comment by lukaseder** [comment]
It seems there is a join predicate missing in the jOOQ meta query, comparing `referential_constraints.table_name = key_column_usage.table_name`. Interesting that this issue hasn't been brought up earlier. It seems that having a foreign key by the same name as a unique key didn't happen too often in the past.

**Comment by lukaseder** [comment]
Fix implemented in jOOQ 3.12. Backport scheduled for 3.11.10 (#8318) and 3.10.9 (#8319).

Thanks again for your report!
